container.go

```diff
@@ -188,7 +190,17 @@ func (c *ContainerRequest) GetContext() (io.Reader, error) {
 	}
 	c.Context = abs
 
-	buildContext, err := archive.TarWithOptions(c.Context, &archive.TarOptions{})
+	// based on https://github.com/docker/cli/blob/master/cli/command/image/build/dockerignore.go#L14
+	var excluded []string
+	if f, err := os.Open(filepath.Join(abs, ".dockerignore")); err != nil {
+		var err error
+		excluded, err = ignorefile.ReadAll(f)
```

Comment:
To avoid a new dependency, I'd also copy the function in here: https://github.com/moby/patternmatcher/blob/main/ignorefile/ignorefile.go

Probably into the `internal/testcontainersdocker` package.

What are your thoughts on this?

Reply:
Exactly, that's what I think too: it's basically parsing the dockerignore file. Only if you agree, let's copy that, but as you said, it's part of the indirect deps already. Whatever is your preference I'm fine with it.

Reply:
I think I prefer leaving it as is. Because the dependency is small and its easier to track upstream changes this way.